### PR TITLE
fixes to bindActionCreators overloads

### PR DESCRIPTION
**Pull request by cellog**
this adds overloads to `bindActionCreators` and fixes the import

### CODE REVIEW

File: src/bindActionCreators.ts [hunk 1]
```diff
@@ -35,6 +35,25 @@ function bindActionCreator<A extends AnyAction = AnyAction>(
  * function as `actionCreators`, the return value will also be a single
  * function.
  */
+export default function bindActionCreators<A, C extends ActionCreator<A>>(
```
Comment (timdorr) [inline src/bindActionCreators.ts]: Please don't use single character generic names. I haven't gone through the existing types to fix this, but I want to start enforcing it now. Preferably via a linting rule.

Comment (nickmccurdy) [reply src/bindActionCreators.ts]: I thought this was the de facto standard for naming generics

Comment (cellog) [reply src/bindActionCreators.ts]: I agree with @timdorr, descriptive names are better if the generic is longer than 1 line.

As for this PR, it is a copy/paste from the existing types in index.d.ts (https://github.com/reduxjs/redux/blob/master/index.d.ts#L559)

How would you feel about doing a follow-up PR that renames everything once we've got the whole thing converted to TS? Suggested steps:

1. make it work as-is, with as few changes as possible
2. merge in #3524 (this is good to go, and will be much harder to rebase after the template variable renames)
3. rename all the things

Comment (markerikson) [reply src/bindActionCreators.ts]: Yep!  I definitely want us to do incremental, history-traceable, step-by-step changes throughout this whole task.  And yes, I'm on board with making types more clearly named.

**Pull request merged**